src/goto-instrument/cover.cpp

```diff
@@ -54,6 +54,33 @@ class basic_blockst
 #else
         it->is_goto() || it->is_function_call();
 #endif
+      // does the basic block contain user code or just our instrumentation?
+      goto_programt::instructionst::const_iterator it_inner=it;
+      if(it_inner!=_goto_program.instructions.end())
+        it_inner++;
+      bool instrumentation=true;
+      while(!it_inner->is_target() &&
```

Comment:
Must check for end() first, otherwise these other conditions read through an invalid iterator